debugger/src/debugger.rs

```diff
@@ -604,212 +601,229 @@ impl Debugger {
+                                    debug_adapter.clear_breakpoint(&mut core_data, request)
+                                }
+                                "show_cpu_register_values" => {
+                                    debug_adapter.show_cpu_register_values(&mut core_data, &request)
+                                }
+                                "dump_cpu_state" => {
+                                    debug_adapter.dump_cpu_state(&mut core_data, &request)
+                                }
+                                "configuration_done" => {
+                                    debug_adapter.configuration_done(&mut core_data, request)
+                                }
+                                "threads" => debug_adapter.threads(&mut core_data, request),
+                                "restart" => debug_adapter.restart(&mut core_data, Some(request)),
+                                "set_breakpoints" => {
+                                    debug_adapter.set_breakpoints(&mut core_data, request)
+                                }
+                                "stack_trace" => debug_adapter.stack_trace(&mut core_data, request),
+                                "scopes" => debug_adapter.scopes(&mut core_data, request),
+                                "source" => debug_adapter.source(&mut core_data, request),
+                                "variables" => debug_adapter.variables(&mut core_data, request),
+                                "continue" => debug_adapter.r#continue(&mut core_data, request),
+                                other => {
                                     debug_adapter.send_response::<()>(
-                                        &request,
-                                        Err(DebuggerError::Other(anyhow!("{}", error))),
-                                    );
-                                    return DebuggerStatus::ErrorTerminateSession;
+                                    request,
+                                    Err(DebuggerError::Other(anyhow!("Received request '{}', which is not supported or not implemented yet", other))),
+                                )?;
+                                    Ok(())
+                                }
+                            };
+
+                            if unhalt_me {
+                                match core_data.target_core.run() {
+                                    Ok(_) => debug_adapter.last_known_status = CoreStatus::Running,
+                                    Err(error) => {
+                                        debug_adapter.send_error_response(
+                                            &DebuggerError::Other(anyhow!("{}", error)),
+                                        )?;
+                                        return Err(error.into());
+                                    }
                                 }
                             }
+
+                            match command_status {
```

Comment:
The code will never get here if command_status is an Err(), because line 766 uses the `?` and will exit the function.   I suggest moving the code from this `match` to replace the `let command_status` on line 731. 

Also, I think it is necessary to return one of the `DebuggerStatus` enum values, or else the behaviour in the loop of `debug_session()` (line 1217) will be broken.

Reply:
Line 766 should only be called for unknown commands, when no command matches. The formatting is a bit weird, but it is part of the `other` branch of the match only, if I understand it correctly.

Reply:
You are correct :-)